pkg/api/alerting/v2alpha1/types_test.go

```diff
@@ -0,0 +1,360 @@
+		State:    rules.StateFiring.String(),
+		Value:    `namespace:workload_cpu_usage:sum{namespace="test"} > 1`,
+		Annotations: map[string]string{
+			"alias":       "The alias is here",
+			"description": "The description is here",
+		},
+		Labels: map[string]string{
+			"alias":       "The alias is here",
+			"description": "The description is here",
+		},
+		ActiveAt: &time.Time{},
+	}
+
+	alerts = []*Alert{
+		alert,
+		{
+			RuleId:   "ca7f09e76954e699",
+			RuleName: "test-cpu-usage-high-2",
+			State:    rules.StateFiring.String(),
+			Value:    `namespace:workload_cpu_usage:sum{namespace="test"} > 1`,
+			Annotations: map[string]string{
+				"alias":       "The alias is here",
+				"description": "The description is here",
+			},
+			Labels: map[string]string{
+				"alias":       "The alias is here",
+				"description": "The description is here",
+			},
+			ActiveAt: &time.Time{},
+		},
+	}
+)
+
+func TestAlertingRuleQueryParams_Filter(t *testing.T) {
+	// empty
+	// Alert Array is empty
+	loacalAlerts := []*Alert{}
+	queryParam := &AlertQueryParams{}
+	ret := queryParam.Filter(loacalAlerts)
+	assert.Empty(t, ret)
+
+	// Alert Array has nil
+	loacalAlerts = []*Alert{nil}
+	ret = queryParam.Filter(loacalAlerts)
+	assert.Empty(t, ret)
+
+	// all pass
+	loacalAlerts = alerts
+	ret = queryParam.Filter(loacalAlerts)
+	assert.NotEmpty(t, ret)
+}
+
+func TestAlertingRuleQueryParams_matches(t *testing.T) {
+	loacalAlert := alert
+	queryParam := alertQueryParams
+	queryParam.LabelEqualFilters = map[string]string{
+		"flags":   "The flags is here",
+		"cpu_num": "The cpu_num is here",
+	}
+	ret := queryParam.matches(loacalAlert)
+	assert.False(t, ret)
+
+	queryParam.LabelEqualFilters = map[string]string{
+		"alias":       "The alias is here",
+		"description": "The description is here",
+	}
+	ret = queryParam.matches(loacalAlert)
+	assert.True(t, ret)
+}
+
+func TestAlertingRuleQueryParams_Sort(t *testing.T) {
+	loacalAlerts := alerts
+	queryParam := alertQueryParams
+	queryParam.Sort(loacalAlerts)
+}
+
+func TestAlertingRuleQueryParams_Sub(t *testing.T) {
+	loacalAlerts := alerts
+	queryParam := alertQueryParams
+	queryParam.Sub(loacalAlerts)
+}
+
+var (
+	succBulkItemResponse = BulkItemResponse{
+		RuleName:  "test-cpu-usage-high",
+		Status:    StatusSuccess,
+		Result:    ResultCreated,
+		ErrorType: "",
+		Error:     nil,
+		ErrorStr:  "",
+	}
+
+	errBulkItemResponse = BulkItemResponse{
+		RuleName:  "test-mem-usage-high",
+		Status:    StatusError,
+		Result:    ResultUpdated,
+		ErrorType: ErrBadData,
+		Error:     errors.New(string(ErrBadData)),
+		ErrorStr:  string(ErrBadData),
+	}
+
+	bulkResponse = BulkResponse{
+		Items: []*BulkItemResponse{
+			&succBulkItemResponse,
+			&errBulkItemResponse,
+		},
+	}
+)
+
+func TestParseAlertingRuleQueryParams(t *testing.T) {
+	queryParam := "/alerting.kubesphere.io/v2alpha1/rules?name=test-cpu&state=firing&health=ok&sort_field=lastEvaluation&sort_type=desc&label_filters=name~test"
```

Comment:
`/alerting.kubesphere.io/v2alpha1/rules` is the request path (relative paths may omit leading slash).

> An HTTP URL takes the form:
> http://<host>:<port>/<path>?<searchpart>
> where <host> and <port> are as described in Section 3.1. If :<port>
is omitted, the port defaults to 80.  No user name or password is
allowed.  <path> is an HTTP selector, and <searchpart> is a query
string. The <path> is optional, as is the <searchpart> and its
preceding "?". If neither <path> nor <searchpart> is present, the "/"
may also be omitted.

https://www.rfc-editor.org/rfc/rfc1738#section-3.3

Reply:
Right! Thanks for your review, I have fixed it 。